src/core/mount-setup.c

```diff
@@ -408,6 +409,8 @@ int mount_setup(bool loaded_policy) {
         mkdir_label("/run/systemd", 0755);
         mkdir_label("/run/systemd/system", 0755);
         mkdir_label("/run/systemd/inaccessible", 0000);
+        touch_file("/run/systemd/inaccessible_file", false, USEC_INFINITY, 0, 0, 0000);
+        label_fix("/run/systemd/inaccessible_file", false, false);
```

Comment:
@keszybz if a process writes to such files (assuming CAP_DAC_OVERRIDE), this would not grow the file on disk unnecessarily.

It is also what Docker does to make files unavailables:
```
$ sudo docker run ubuntu ls -l /proc/kcore
crw-rw-rw-. 1 root root 1, 3 May 25 14:56 /proc/kcore
$ ls -l /dev/null 
crw-rw-rw-. 1 root root 1, 3 May 21 19:47 /dev/null
```